compiler/rustc_typeck/src/collect.rs

```diff
@@ -3267,6 +3268,24 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
     codegen_fn_attrs
 }
 
+/// Computes the set of target features used in a function for the purposes of
+/// inline assembly.
+fn asm_target_features<'tcx>(tcx: TyCtxt<'tcx>, id: DefId) -> &'tcx FxHashSet<Symbol> {
+    let mut target_features = tcx.sess.target_features.clone();
+    let attrs = tcx.codegen_fn_attrs(id);
```

Comment:
I think one problem that this has is that `-Ctarget-feature` flag attributes are not accounted for here, to the best of my knowledge. This is something we should eventually fix (at the very least because `#[cfg(target_feature=...)]` honour the global flags.) One of the commits in https://github.com/rust-lang/rust/pull/87402 gets us partway towards a solution there by exposing the globally enabled features as a query, but those are still backend-specific names so not applicable for assembly quite yet.